src/status_im/contexts/onboarding/events.cljs

```diff
@@ -89,10 +266,14 @@
   {:events [:onboarding/on-delete-profile-success]}
   [{:keys [db]} key-uid]
   (let [multiaccounts (dissoc (:profile/profiles-overview db) key-uid)]
-    (merge
-     {:db (assoc db :profile/profiles-overview multiaccounts)}
-     (when-not (seq multiaccounts)
-       {:dispatch [:update-theme-and-init-root :screen/onboarding.intro]}))))
+    {:db (assoc db :profile/profiles-overview multiaccounts)
+     :fx (cond-> []
+           (ff/enabled? ::ff/settings.news-notifications)
+           (conj [:effects.profile/remove-local-profile-storage
+                  {:key-uid key-uid}])
+           (not (seq multiaccounts))
+           (conj [:dispatch
+                  [:update-theme-and-init-root :screen/onboarding.intro]]))}))
```

Comment:
Here we're extending the `:onboarding/on-delete-profile-success` event to remove any profile related data that had been stored inside the `mmkv` storage via the `key-uid`.